tests/constbuffer_array_ut/constbuffer_array_ut.c

```diff
@@ -813,6 +827,616 @@ TEST_FUNCTION(constbuffer_array_create_from_buffer_index_and_count_fails_when_un
+TEST_FUNCTION(constbuffer_array_create_from_buffer_index_and_count_succeeds_only_get_part_of_start_buffer)
+{
+    // arrange
+    CONSTBUFFER_HANDLE test_buffers[4];
+    CONSTBUFFER_ARRAY_HANDLE original;
+    CONSTBUFFER_ARRAY_HANDLE constbuffer_array;
+
+    test_buffers[0] = TEST_CONSTBUFFER_HANDLE_1;
+    test_buffers[1] = TEST_CONSTBUFFER_HANDLE_2;
+    test_buffers[2] = TEST_CONSTBUFFER_HANDLE_3;
+    test_buffers[3] = TEST_CONSTBUFFER_HANDLE_4;
+
+    original = constbuffer_array_create(test_buffers, sizeof(test_buffers) / sizeof(test_buffers[0]));
+    umock_c_reset_all_calls();
+
+    constbuffer_array_create_from_start_and_end_inert_path(1, 1, 1, 0, 4);
+
+    // act
+    constbuffer_array = constbuffer_array_create_from_start_and_end(original, 1, 3, 1, 4);
+
+    // assert
+    ASSERT_IS_NOT_NULL(constbuffer_array);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    uint32_t actual_buffer_count;
+    (void)constbuffer_array_get_buffer_count(constbuffer_array, &actual_buffer_count);
+    ASSERT_ARE_EQUAL(uint32_t, 3, actual_buffer_count);
+    const CONSTBUFFER_HANDLE* internal_array = constbuffer_array_get_const_buffer_handle_array(constbuffer_array);
+    CONSTBUFFER_HANDLE start_buffer = real_CONSTBUFFER_CreateFromOffsetAndSize(TEST_CONSTBUFFER_HANDLE_2, 1, 1);
+
+    ASSERT_ARE_EQUAL(bool, true, real_CONSTBUFFER_HANDLE_contain_same(start_buffer, internal_array[0]));
+    ASSERT_ARE_EQUAL(void_ptr, TEST_CONSTBUFFER_HANDLE_3, internal_array[1]);
+    ASSERT_ARE_EQUAL(void_ptr, TEST_CONSTBUFFER_HANDLE_4, internal_array[2]);
+
+    // cleanup
+    real_CONSTBUFFER_DecRef(start_buffer);
+    constbuffer_array_dec_ref(original);
+    constbuffer_array_dec_ref(constbuffer_array);
+}
+
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_007: [ constbuffer_array_create_from_start_and_end shall allocate memory for a new CONSTBUFFER_ARRAY_HANDLE. ]*/
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_011: [ constbuffer_array_create_from_start_and_end shall compute the start buffer size. ]*/
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_005: [ constbuffer_array_create_from_start_and_end shall get the start buffer by calling CONSTBUFFER_CreateFromOffsetAndSize. ]*/
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_012: [ constbuffer_array_create_from_start_and_end shall get the end buffer by calling CONSTBUFFER_CreateFromOffsetAndSize. ]*/
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_008: [ constbuffer_array_create_from_start_and_end shall copy all of the CONSTBUFFER_HANDLES except first and last buffer from each const buffer array in buffer_arrays to the newly constructed array by calling CONSTBUFFER_IncRef. ]*/
+/* Tests_SRS_CONSTBUFFER_ARRAY_07_009: [ constbuffer_array_create_from_start_and_end shall return a non - NULL handle. ]*/
+TEST_FUNCTION(constbuffer_array_create_from_buffer_index_and_count_succeeds_only_get_part_of_start_buffer_0_start_buffer_size)

```

Comment:
>_0_start_buffer_size [](http://example.com/codeflow?start=105&length=20)

not sure what this part means?

Reply:
nevermind I get it now, the first buffer will have 0 size because offset is the same as the buffer size